### PR TITLE
Feature/retrieve value as string 1

### PR DESCRIPTION
**Pull request by tamerharsh**

### CODE REVIEW

File: commandlineparser/src/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #CMD_PARASER_LIB/SRC 
 include_directories("${PROJECT_SOURCE_DIR}/commandlineparser/include")
 
-add_library(mylib Options.cpp Parser.cpp)
+add_library(mylib Options.cpp Parser.cpp  "../include/Parser.h" "../include/Options.h")
```
Comment (inbangsa) [inline commandlineparser/src/CMakeLists.txt:4]: Its not a good idea to do this in this branch..  a bugfix would be a good idea.. No need to change this now.. But keep this as you learnings

File: commandlineparser/src/Parser.cpp [hunk 2]
```diff
@@ -130,3 +137,85 @@ void cmdParser::Parser::long_help(const std::vector<std::string>&keys) const
...
+			{
+				std::string temp(argv[i]);
+				size_t found = temp.find("=");
+				if (found != std::string::npos)
+				{
+					key = temp.substr(0, found);
+					std::string val = temp.substr(temp.find("=") + 1);
+
+					//validity checking block.
+					if (command_list.find(key) == command_list.end())
```
Comment (inbangsa) [inline commandlineparser/src/Parser.cpp]: optimize this.. same condition check two times

File: commandlineparser/src/Parser.cpp [hunk 3]
```diff
@@ -130,3 +137,85 @@ void cmdParser::Parser::long_help(const std::vector<std::string>&keys) const
...
+	}
+}
+
+std::vector <std::string> cmdParser::Parser::getValueAsString(const std::string &  input)const
+{ 
+	//retrive value if the input key is found in command list.
+	if (command_list.find(input) != command_list.end())
+	{
+		// demanding value for command for which user has not given any value from commandline.(later we will set it to default).
+		if (command_list.at(input)->get_value().empty())
```
Comment (inbangsa) [inline commandlineparser/src/Parser.cpp]: Optimize this.. call get_value() only once

**Review by inbangsa — CHANGES_REQUESTED** [review]

File: commandlineparser/src/CMakeLists.txt (hunk 1 above)
Comment (tamerharsh) [reply commandlineparser/src/CMakeLists.txt:4]: ok.

File: commandlineparser/src/Parser.cpp (hunk 2 above)
Comment (tamerharsh) [reply commandlineparser/src/Parser.cpp]: done.

File: commandlineparser/src/Parser.cpp (hunk 3 above)
Comment (tamerharsh) [reply commandlineparser/src/Parser.cpp]: done.

**Comment by inbangsa** [comment]
Not needed

**Pull request closed without merging**